tests/Unit/Domain.LinnApps.Tests/PurchaseOrderServiceTests/WhenCreatingFOCPurchaseOrder.cs

```diff
@@ -0,0 +1,191 @@
+                                 OrderContactName = "Rick.F.Pile",
+                                 OrderMethod = new OrderMethod { Name = "online", Description = "website" },
+                                 ExchangeRate = 0.8m,
+                                 IssuePartsToSupplier = "N",
+                                 DeliveryAddress = new LinnDeliveryAddress { AddressId = 1555 },
+                                 RequestedBy = new Employee { FullName = "Employee O'RealPerson", Id = 1111 },
+                                 EnteredBy = new Employee { FullName = "Person McPerson", Id = 2222 },
+                                 QuotationRef = "ref11101",
+                                 AuthorisedBy = new Employee { FullName = "Mr Manager", Id = 3333 },
+                                 SentByMethod = "EMAIL",
+                                 FilCancelled = string.Empty,
+                                 Remarks = "updated remarks",
+                                 DateFilCancelled = null,
+                                 PeriodFilCancelled = null,
+                                 DocumentType = new DocumentType { Description = "Purchase order", Name = "PO" }
+                             };
+
+            this.MockAuthService.HasPermissionFor(AuthorisedAction.PurchaseOrderCreate, Arg.Any<IEnumerable<string>>())
+                .Returns(true);
+
+            this.PurchaseOrdersPack.GetVatAmountSupplier(Arg.Any<decimal>(), Arg.Any<int>()).Returns(40.55m);
+
+            this.MockDatabaseService.GetIdSequence("PLORP_SEQ").Returns(123);
+
+            this.PurchaseLedgerMaster.GetRecord().Returns(new PurchaseLedgerMaster { OkToRaiseOrder = "Y" });
+
+            this.PartSupplierRepository.FindById(Arg.Any<PartSupplierKey>()).Returns(
+                new PartSupplier { UnitOfMeasure = "Carrots", LeadTimeWeeks = 2 });
+
+            this.result = this.Sut.CreateOrder(this.order, new List<string>());
+        }
+
+        [Test]
+        public void ShouldSetFields()
+        {
+            this.result.SentByMethod.Should().Be("EMAIL");
+
+            var firstDetail = this.result.Details.First();
+
+            firstDetail.OrderPosting.Id.Should().Be(123);
+            firstDetail.OrderConversionFactor.Should().Be(1, "conversion factor set to 1 for everything for now");
+
+            firstDetail.OurQty.Should().Be(99m);
+
+            // updated based on conv factor
+            firstDetail.OrderQty.Should().Be(99m);
+
+            firstDetail.OurUnitPriceCurrency.Should().Be(0m);
+
+            // updated based on conv factor
+            firstDetail.OrderUnitPriceCurrency.Should().Be(0m);
+
+            firstDetail.BaseOurUnitPrice.Should().Be(0m);
+
+            // our qty * our unit price
+            firstDetail.NetTotalCurrency.Should().Be(0m);
+            firstDetail.BaseNetTotal.Should().Be(0m);
+
+            firstDetail.VatTotalCurrency.Should().Be(40.55m);
```

Comment:
how can the order have non zero totals if its free of charge?

Reply:
good point

Reply:
I don't think the code is wrong, I think its just because your mock data has some non zero values. I'd make them all zero for clarity.

Reply:
Yeah, I wasn't entirely sure if I should have kept the base VAT rates used in the other PO forms. Changed the set up so the PurchaseOrderPack now returns a VAT of 0 and keeps the totals to 0.

Reply:
you could also test the created delivery's monetary values get zeroed too